x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlRealmTests.java

```diff
@@ -534,7 +556,96 @@ public SamlRealm buildRealm(
         }
     }
 
-    public void testAttributeSelectionWithRegex() throws Exception {
+    public void testAttributeSelectionWithSplit() {
+        String delimiter = ",";
+
+        final Settings settings = Settings.builder()
+            .put(REALM_SETTINGS_PREFIX + ".attributes.groups", "departments")
+            .put(REALM_SETTINGS_PREFIX + ".attribute_delimiters.groups", delimiter)
+            .build();
+
+        final RealmConfig config = buildConfig(settings);
+
+        final SamlRealmSettings.PatternDelimiterAttributeSetting groupSetting = new SamlRealmSettings.PatternDelimiterAttributeSetting(
+            "groups"
+        );
+        final SamlRealm.AttributeParser parser = SamlRealm.AttributeParser.forSetting(logger, groupSetting, config, false);
+
+        final SamlAttributes attributes = new SamlAttributes(
+            new SamlNameId(NameIDType.TRANSIENT, randomAlphaOfLength(24), null, null, null),
+            randomAlphaOfLength(16),
+            Collections.singletonList(
+                new SamlAttributes.SamlAttribute(
+                    "departments",
+                    "departments",
+                    Collections.singletonList(String.join(delimiter, "engineering", "elasticsearch-admins", "employees"))
+                )
+            )
+        );
+
+        final List<String> strings = parser.getAttribute(attributes);
+        assertThat("For attributes: " + strings, strings, contains("engineering", "elasticsearch-admins", "employees"));
+    }
+
```

Comment:
Let's also add a validation test where we try to specify `attribute_delimiters` for an attribute other than `groups`.

Reply:
I'm also thinking it might make for better UX to add a more specific failure message, like "splitting an attribute on a delimiter is only supported for the groups attribute, but found setting for ... please remove ..."

Reply:
The node won't start if we do that since the setting isn't registered. It logs:
```
unknown setting [xpack.security.authc.realms.saml.saml1.attribute_delimiter.principal] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
``` 
Is there a way to hook in to that settings validation from within the security plugin?